main/main.js

```diff
@@ -280,6 +280,11 @@ ipc.on('showSecondaryMenu', function (event, data) {
   }
 })
 
+ipc.on('proxyChange', function(event, data) {
+  session.defaultSession.setProxy(data)
```

Comment:
Private tabs use a different session then the default, so if we do this, anything you load in a private tab won't be proxied. I'm not actually sure what the best way to get a list of all the current sessions is, I'm guessing viewManager.js would need to be modified to keep a list of them. I'll try to think about it more and see if I can come up with a solution.

Reply:
The main reason I used ipc listener was that I couldn't have access to session anywhere else rather than main class, for private sessions we have to do the same this as data is the same proxy string here.

Reply:
> The main reason I used ipc listener was that I couldn't have access to session anywhere else rather than main class,

I think you can just move the entire `proxy.js` script to run in the main process; that would let you use the session module without an IPC message.

> for private sessions we have to do the same this as data is the same proxy string here.

Right, I'm just not sure how to get the `session` object that represents each private tab.